sktime/forecasting/trend.py

```diff
@@ -212,3 +214,280 @@ def _predict(self, fh=None, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+            else clone(self.forecaster_resid)
+        )
+
+        # fitting forecasters to different components
+        self.forecaster_seasonal_.fit(y=self.seasonal_, X=X, fh=fh)
+        self.forecaster_trend_.fit(y=self.trend_, X=X, fh=fh)
+        self.forecaster_resid_.fit(y=self.resid_, X=X, fh=fh)
+
+    def _predict(self, fh, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+        """Forecast time series at future horizon.
+
+        Parameters
+        ----------
+        fh : int, list, np.array or ForecastingHorizon
+            Forecasting horizon
+        X : pd.DataFrame, optional (default=None)
+                Exogenous time series
+        return_pred_int : bool, optional (default=False)
+            If True, returns prediction intervals for given alpha values.
+        alpha : float or list, optional (default=0.95)
+
+        Returns
+        -------
+        y_pred : pd.Series
+            Point predictions
+        """
+        y_pred_seasonal = self.forecaster_seasonal_.predict(fh=fh, X=X)
+        y_pred_trend = self.forecaster_trend_.predict(fh=fh, X=X)
+        y_pred_resid = self.forecaster_resid_.predict(fh=fh, X=X)
+        y_pred = y_pred_seasonal + y_pred_trend + y_pred_resid
+        return y_pred
+
+    def _update(self, y, X=None, update_params=True):
+        """Update cutoff value and, optionally, fitted parameters.
+
+        Parameters
+        ----------
+        y : pd.Series, pd.DataFrame, or np.array
+            Target time series to which to fit the forecaster.
+        X : pd.DataFrame, optional (default=None)
+            Exogeneous data
+        update_params : bool, optional (default=True)
+            whether model parameters should be updated
+
+        Returns
+        -------
+        self : reference to self
+        """
+        # TODO: remove _update method after refactor of update methods in base class.
+        # This method is only required to pass tests.
+        self.stl_ = _STL(
+            y.values,
+            period=self.sp,
+            seasonal=self.seasonal,
+            trend=self.trend,
+            low_pass=self.low_pass,
+            seasonal_deg=self.seasonal_deg,
+            trend_deg=self.trend_deg,
+            low_pass_deg=self.low_pass_deg,
+            robust=self.robust,
+            seasonal_jump=self.seasonal_jump,
+            trend_jump=self.trend_jump,
+            low_pass_jump=self.low_pass_jump,
+        ).fit()
+
+        self.seasonal_ = pd.Series(self.stl_.seasonal, index=y.index)
+        self.resid_ = pd.Series(self.stl_.resid, index=y.index)
+        self.trend_ = pd.Series(self.stl_.trend, index=y.index)
+
+        self.forecaster_seasonal_.update(
+            y=self.seasonal_, X=X, update_params=update_params
+        )
+        self.forecaster_trend_.update(y=self.trend_, X=X, update_params=update_params)
+        self.forecaster_resid_.update(y=self.resid_, X=X, update_params=update_params)
+        return self
```

Comment:
Isn't `_update` optional for forecasting?

Reply:
there is an issue with default `_update`, it was not working correctly. We need to refactor that anyways. What I do here is excactly refitting again, the `STL` as well as forecasters, its the same behavior as `_fit()`, unless I made a mistake?

Reply:
Ok - in that case replace `update` in lines 488-492 with `fit`. This way it will 100% refit. Now it's refitting STL and updating the season, trend, residuals forecasters instead of refitting these too.

Reply:
I think its fine to `update` the forecasters if as they might have implemented a custom `_update`. The `STL` does not have an `update` so we need to refit it. Actually I think how I have implemented is even better than using the default `_update` from base class (if it would work 😆 )

Reply:
It might be fine, but I'm not sure. When you refit STL you generate 3 new time series for the same time index as before. So if you then `update` instead of `fit`, you're not really updating in the sense that you got a few more time steps in your time series. You're updating your estimator with a completely new time series, which is why it makes more sense to refit instead (at least to me). What do you think?